clean clean clean :D (#1880)
enkidevs/curriculum: javascript/core/intro-js/javascript-basics-part-2.md

---
## Content

> Like most programming languages, JavaScript has a number of syntax rules that it must adhere to.

Author: ```suggestion
Like most programming languages, JavaScript has a number of syntax rules that it must adhere to.
```

### Semicolon

> Statement in JavaScript end in a `;`. Based on the ECMAScript[1] rules, adding a `;` isn't actually required (but is common practice):

Author: ```suggestion
Statements in JavaScript end in a `;`. Based on the ECMAScript[1] rules however, adding a `;` isn't actually required (although it is common practice):
```

> Statements in JavaScript end in a `;`. Based on the ECMAScript[1] rules however, adding a `;` isn't actually required (although it is common practice):

Reviewer: Just a suggestion.
`Isn't actually required` can be shortened to just `isn't required`.

```javascript
let firstName = "Enki";

> //                  ^

Author: ```suggestion
//                    ^
```

```

Omitting a semicolon in certain situations can cause problems. JavaScript has a feature called Automatic Semicolon Insertion (ASI) which means that, if you omit a semicolon, JavaScript will automatically add it where it thinks it should go. 

> This can sometimes lead to unexpected results[2], and it's usually recommended to always insert the `;` yourself.

Author: ```suggestion
This can sometimes lead to unexpected results[2], thus it's usually recommended to always insert the `;` yourself.
```

> This can sometimes lead to unexpected results[2], thus it's usually recommended to always insert the `;` yourself.

Reviewer 2: what do you think of this?
```suggestion
This can sometimes lead to unexpected results[2]. Thus, it's usually recommended to insert the `;` yourself.
```
